### PR TITLE
[FAQ/ComparisonTable] Adjust heading space

### PR DESCRIPTION
**Pull request by josepmartins**
### <samp>🤖 Generated by Copilot at 9e9175b</samp>

This pull request updates the FAQ and ComparisonTable components to use design tokens for consistent and responsive styling. It adds new files for the design tokens in `packages/design-tokens` and imports them in the corresponding CSS and TypeScript files in `packages/react`. It also removes some unnecessary or conflicting CSS rules and code.

This works is a follow up on this PR from @egim: https://github.com/primer/brand/pull/241 
- Updated margin spacing for the ComparisonTable heading and the FAQ based on a [design feedback issue](https://github.com/github/marketing-engineering/issues/1397). This inner margins values also reacts to the screen size and it's values are `96px` on large devices, `64px` on medium ones and `40px` in small ones. 
- `ComparisonTable` keeps the `8px` margin on narrow screens as an special treatment.
- Added a max-width property on headings using `ch` values. This is similar to what we do on `Pillar` or `SectionIntro` components.

This changes are being upstreamed to the Figma library through this branch: https://www.figma.com/file/0WFyt65XXKvDXZab2wBTUi/josepmartins%2Ffaq-comparisontable-heading?type=design&node-id=3362-30140&mode=design

## List of notable changes:

- **added** responsive tokens for inner spacing in `FAQ` and `ComparisonTable` headings.
- **added** max-width properties in `FAQ` and `ComparisonTable` headings

## What should reviewers focus on?

- FAQ Storybook examples that uses headings in both components must have `40px` margin separation on small screens , `64px` in medium and `96px` in large ones. 
- Max-width should be `18ch` for headings in both components

## Steps to test:

1. Open the # component in CI-deployed preview environment
2. Go to # story in Storybook
3. Verify that # behaves as described in the following issue.

## Supporting resources (related issues, external links, etc):

- Design issue: https://github.com/primer/brand/pull/241
- Initial PR: https://github.com/primer/brand/pull/241

## Contributor checklist:

- [ ] All new and existing CI checks pass
- [ ] Tests prove that the feature works and covers both happy and unhappy paths
- [ ] Any drop in coverage, breaking changes or regressions have been documented above
- [ ] New visual snapshots have been generated / updated for any UI changes
- [ ] All developer debugging and non-functional logging has been removed
- [ ] Related issues have been referenced in the PR description

## Reviewer checklist:

- [ ] Check that pull request and proposed changes adhere to our [contribution guidelines](../../CONTRIBUTING.md) and [code of conduct](../../CODE_OF_CONDUCT.md)
- [ ] Check that tests prove the feature works and covers both happy and unhappy paths
- [ ] Check that there aren't other open Pull Requests for the same update/change

## Screenshots:

> Please try to provide before and after screenshots or videos

<table>
<tr>
<th> Before</th> <th> After </th>
</tr>
<tr>
<td valign="top">

 </td>
<td valign="top">

</td>
</tr>
</table>

### CODE REVIEW

File: packages/react/src/ComparisonTable/ComparisonTable.module.css [hunk 1]
```diff
@@ -3,12 +3,17 @@
 */
 .ComparisonTable--heading {
   text-align: left;
+  max-width: 18ch;
+  margin-bottom: var(--brand-ComparisonTable-spacing-inner-heading);
```
Comment (rezrah) [inline packages/react/src/ComparisonTable/ComparisonTable.module.css:7]: I think heading could be brought forward a little, as it's part of the pattern and could help leave door open to co-locate additional tokens.

WDYT about `--brand-ComparisonTable-heading-marginBottom`

☝️ This feels more targeted. @lukasoppermann just confirmed that marginBottom is acceptable too, which is neat.

Comment (josepmartins) [reply packages/react/src/ComparisonTable/ComparisonTable.module.css]: Sounds good, I've made the changes here: https://github.com/primer/brand/pull/313/commits/2f87494bbbcdc2ff4de6c9e5cc96db15cfc38111

File: packages/react/src/FAQ/FAQ.tsx [hunk 2]
```diff
@@ -52,7 +53,6 @@ const FAQRoot = forwardRef<HTMLElement, FAQRootProps>(({children, style, animate
             return React.cloneElement(child as React.ReactElement, {
               align: hasSubheading ? 'start' : child.props.align,
               size: hasSubheading ? 'large' : child.props.size,
-              className: clsx(!hasSubheading && styles['FAQ__heading--with-margin'], child.props.className),
```
Comment (rezrah) [inline packages/react/src/FAQ/FAQ.tsx:55]: @josepmartins - should this have been removed?

**Review by rezrah — APPROVED** [review]
Looks good. Just need to update the snapshots again.

**Pull request merged**